ckan/lib/mailer.py

```diff
@@ -141,7 +141,8 @@ def get_reset_link(user):
                    h.url_for(controller='user',
                              action='perform_reset',
                              id=user.id,
-                             key=user.reset_key))
+                             key=user.reset_key,
+                             qualified=True))
```

Comment:
Just passing `qualified=True` is not enough, since the code still joins the result with `site_url`. The latter is wrong and needs to be removed. All you should need is the result of `url_for` (with `qualified=True`).